charts/community-operator/templates/mongodbcommunity_cr_with_tls.yaml

```diff
@@ -74,9 +128,16 @@ spec:
         name: {{ .Values.resource.tls.caCertificateSecretRef }}
       {{- end }}
     authentication:
+      {{- if .Values.resource.tls.useX509 }}
+      modes: ["X509"]
+      {{- else }}
       modes: ["SCRAM"]
-  {{- with .Values.resource.users }}
+      {{- end }}
+  {{- if .Values.resource.tls.sampleX509User }}
```

Comment:
I am not sure whether I like the semantics of this. We are basically saying that if `sampleX509User` then we apply all `x509Users` which contains a sampleUser.

But what if i configured my own x509 users, does this mean I have to activate `sampleX509User` as well?

Reply:
If you were to add your own x509 users, you shouldn't activate that flag. That flag represents a way to just generate a sample user. Indeed, if that flag is activated, then your users will be ignored, and only the sample will be used. This flag also creates a certificate for that sample user, so it would be used mainly for testing purposes, but it also serves as an example for how to create x509 users and certificates.